src/services/jsonforms.ts

```diff
@@ -355,10 +368,13 @@ const generateUISchema = (
     }
 
     // If we've gotten here, then the schema appears to be for a scalar value. For most of these, we
-    // just create a default Control, but for string types, we set the `multi` option based on
-    // whether the json schema contains a `multiline` annotation.
+    // just create a default Control, but we first check if we should turn the input into a password
+    // input. This overrides other settings to keep things secure. Then we check if we need to set
+    // the `multi` option based on whether the json schema contains a `multiline` annotation.
     const controlObject: ControlElement = createControlElement(currentRef);
-    if (isMultilineText(jsonSchema)) {
+    if (isSecretText(jsonSchema)) {
+        addOption(controlObject, 'format', 'password');
+    } else if (isMultilineText(jsonSchema)) {
         addOption(controlObject, 'multi', true);
     }
```

Comment:
Decided to make the check for secret first as that should never be overwritten

Reply:
I'm curious whether these two options need to be mutually exclusive. Would it work to have a field that is a multi-line secret? The use case I'm thinking of is for ssh tunneling, where you need to paste in a private key. IIRC it actually posed a problem for that field to _not_ be multi-line because it would strip out all newline characters, which causes the certificate to be malformed. This is the only use case that I know of currently for multi-line secrets, so we can talk though some other options if that seems unworkable.

Reply:
You cannot do that with HTML. So if we ever want a multi line secret then we'll have to build an input for it.

Reply:
Google service accounts may be motivating here. They're both big and also secret.